src/core/radio/radio.hpp

```diff
@@ -769,9 +770,7 @@ inline Error Radio::EnableCsl(uint32_t aCslPeriod, otShortAddress aShortAddr, co
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE || OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
 inline uint8_t Radio::GetCslAccuracy(void) { return otPlatRadioGetCslAccuracy(GetInstancePtr()); }
-#endif
 
-#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```

Comment:
Keep these as before?

Reply:
The idea was that both CSL transmitter and receiver need access to this function. However I see now that CSL receivers (MTD) have the flag `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` also enabled, so we don't need to change here indeed.    The setting is counter-intuitive - is there a specific reason why this flag is also `1` for an MTD-ony build? Since it would never be a CSL transmitter.

Reply:
This is related to RCP/host compatibility. I think we changed it there (plat api is only under certain config), so want us to keep it the same here (in `radio.hpp`)

> The setting is counter-intuitive - is there a specific reason why this flag is also 1 for an MTD-ony build? Since it would never be a CSL transmitter.

I was not myself much involved in CSL development, so don't know if there was a reson.

Reply:
@EskoDijk

> The setting is counter-intuitive - is there a specific reason why this flag is also 1 for an MTD-ony build? Since it would never be a CSL transmitter.

Where is this flag set to 1 for an MTD-only build?

Reply:
> Where is this flag set to 1 for an MTD-only build?

https://github.com/openthread/openthread/blob/main/src/core/config/mac.h#L405

Reply:
@EskoDijk @edmont 

For explanation:

Unlike other configurations that can be defined in core config files, `OPENTHREAD_FTD`, `OPENTHREAD_MTD` and `OPENTHREAD_RADIO` are special and are passed by the build system. Given that some other modules (like third_party) will also include the core config files but they are not built with FTD/MTD/RADIO definitions, this will cause some warnings. Because when building these modules, they will find the `OPENTHREAD_FTD`, `OPENTHREAD_MTD`, `OPENTHREAD_RADIO` are not defined.

So we will avoid using `OPENTHREAD_FTD`, `OPENTHREAD_MTD` and `OPENTHREAD_RADIO` in core config files. We will not use these options to define other options. For example, we won't define `CSL_TRANSMITTER` by `OPENTHREAD_FTD`. Instead, we will use `OPENTHREAD_FTD`, `OPENTHREAD_MTD`, `OPENTHREAD_RADIO` together with the feature wrapper in the code. For example, `#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE`. #5923 did the replacement.